src/main.rs

```diff
@@ -509,11 +495,19 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
     // Probe for hardcoded reset device. Not really a probe.
     ResetDriver::probe_from(&hyp_dt, &mut mem_map).expect("Failed to set up Reset Device");
 
-    setup_hyp_paging(&mut mem_map);
-
     // Set up per-CPU memory and boot the secondary CPUs.
     PerCpu::init(hart_id, &mut mem_map);
 
+    // Create an allocator for the remaining pages. Anything that's left over will be mapped
+    // into the host VM.
+    let mut hyp_mem = HypPageAlloc::new(&mut mem_map);
+
+    // We are about to set up paging based on the hardware memory map. It can't be modified from now on.
+    let mem_map = mem_map; // Remove mutability.
```

Comment:
Nit: The "lousily" in the commit message can be dropped given that this is an accepted Rust idiomatic practice to convert a mutable reference to mutable, and the enforcement is assured by the compiler.

Reply:
Okay. Still the reality of a computer doesn't stop at the compiler. I consider it lousy because something somewhere can still write to it.

Reply:
Done. Not lousily.